bridges/AnidexBridge.php

```diff
@@ -0,0 +1,205 @@
+			}
+		}
+		foreach (array('r', 'b', 'a') as $param_name) {
+			$param = $this->getInput($param_name);
+			if (!empty($param) && boolval($param)) {
+				$search_url .= '&' . $param_name . '=1';
+			}
+		}
+		$query = $this->getInput('q');
+		if (!empty($query)) {
+			$search_url .= '&q=' . urlencode($query);
+		}
+		$opt = array();
+		$h = $this->getInput('h');
+		if (!empty($h) && intval($h) != 0 && ctype_digit($h)) {
+			$opt[CURLOPT_COOKIE] = 'anidex_h_toggle=' . $h;
+		}
+
+		// Retrieve torrent listing from search results, which does not contain torrent description
+		$html = getSimpleHTMLDOM($search_url, array(), $opt)
+		or returnServerError('Could not request Anidex: '.$search_url);
+		$links = $html->find('a');
+		$results = array();
+		foreach ($links as $link)
+			if (strpos($link->href, '/torrent/') === 0 && !in_array($link->href, $results))
+				$results[] = $link->href;
+		if (empty($results))
+			returnServerError('No results from Anidex: '.$search_url);
+		$limit = 0;
+
+		//Process each item individually
+		foreach ($results as $element) {
+
+			$torrent_id = str_replace('/torrent/', '', $element);
+
+			//Limit total amount of requests and ignore entries without valid torrent ID
+			if ($limit < 20 && $torrent_id != 0 && ctype_digit($torrent_id)) {
+
+				//Retrieve data for this torrent ID
+				$item_uri = self::URI . 'torrent/'.$torrent_id;
+
+				//Retrieve full description from torrent page
+				if ($item_html = getSimpleHTMLDOMCached($item_uri)) {
+
+					//Retrieve data from page contents
+					$item_title = str_replace(' (Torrent) - AniDex ', '', $item_html->find('title', 0)->plaintext);
+					$item_desc = $item_html->find('div.panel-body', 0);
+					$item_author = trim($item_html->find('span.fa-user', 0)->parent()->plaintext);
+					$item_date = strtotime(trim($item_html->find('span.fa-clock', 0)->parent()->plaintext));
+					$item_image = $this->getURI().'images/user_logos/default.png';
+
+					//Check for description-less torrent andn optionally extract image
+					$desc_title_found = false;
+					foreach ($item_html->find('h3.panel-title') as $h3) {
+						if (strpos($h3, 'Description') !== false) {
+							$desc_title_found = true;
+							break;
+						}
+					}
+					if ($desc_title_found) {
+						//Retrieve image for thumbnail or generic logo fallback
+						foreach ($item_desc->find('img') as $img) {
+							if (strpos($img->src, 'prez') === false) {
+								$item_image = $img->src;
+								break;
+							}
+						}
+						$item_desc = trim($item_desc->innertext);
+					} else {
+						$item_desc = '<em>No description.</em>';
+					}
+
+					//Build and add final item
+					$item = array();
+					$item['uri'] = $item_uri;
+					$item['title'] = $item_title;
+					$item['author'] = $item_author;
+					$item['timestamp'] = $item_date;
+					$item['enclosures'] = array($item_image);
+					$item['content'] = $item_desc;
+					$this->items[] = $item;
+					$limit++;
```

Comment:
Small improvement: You can get rid of $limit by checking the array size and breaking the loop:

```PHP
if(count($this->items) >= 20) {
    break;
}
```

Reply:
Fixing :+1: